app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+        }
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingHistory -> {
+      ShowHistory(
+        onIconClick = { showBottomSheet = true },
+        onCertificateClick = onCertificateClick,
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
+  onIconClick: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  historyList: List<TravelCertificate>,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    modifier = Modifier.clearFocusOnTap(),
+    topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+    topAppBarActions = {
+      IconButton(
+        onClick = { onIconClick() },
+        modifier = Modifier.size(40.dp),
+      ) {
+        Icon(
+          imageVector = Icons.Hedvig.Info,
+          contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+          modifier = Modifier.size(24.dp),
+        )
+      }
+    },
+  ) {
+    if (historyList.isEmpty()) {
+      Spacer(modifier = Modifier.weight(1f))
+      ShowInitialInfo()
+    } else {
+      ShowNotEmptyList(
+        list = historyList,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+    Spacer(modifier = Modifier.weight(1f))
+    VectorInfoCard(
+      text = String.format(stringResource(id = R.string.travel_certificate_start_date_info), 45),
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp, 16.dp, 16.dp, 8.dp),
+    )
+    HedvigSecondaryContainedButton(
+      text = stringResource(R.string.travel_certificate_get_travel_certificate_button),
+      onClick = onContinue,
+      modifier = Modifier.padding(16.dp, 8.dp, 16.dp, 0.dp),
+    )
+    Spacer(Modifier.height(32.dp))
+  }
+}
+
+@Composable
+private fun ShowInitialInfo() {
+  Column(
+    modifier = Modifier.fillMaxSize(),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center,
+  ) {
+    HedvigInformationSection(
+      stringResource(id = R.string.travel_certificate_empty_list_message),
+    )
+  }
+}
+
+@Composable
+private fun ShowNotEmptyList(list: List<TravelCertificate>, onCertificateClick: (String) -> Unit) {
+  Column(
+    horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    for (i in list) {
+      CertificateCard(
+        certificate = i,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+  }
+}
+
+@Composable
+private fun CertificateCard(
+  certificate: TravelCertificate,
+  onCertificateClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier
+      .fillMaxWidth()
+      .clickable(onClick = { onCertificateClick(certificate.signedUrl) })
+      .padding(16.dp),
+  ) {
+    val isExpired = isExpired(certificate.expiryDate)
```

Comment:
Great opportunity here to think about if this should be happening in the UI or not.
How would you feel if this `isExpired` was exposed directly from the ViewModel, from inside the `UiState` for each item? So if each `TravelCertificate` already had this information instead of having to determine that in the UI itself? This is especially true considering that doing it in the UI moves logic like comparing the expiration date with the current year, and doing a calculation etc.
The ViewModel could instead be taking in a `clock`, exactly as we do [here](https://github.com/HedvigInsurance/android/blob/2dc8dd0e355e2036d888e6e81d35c30e61c676c4/app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/ChatViewModel.kt#L11) for example. That way we can even in the future test this behavior in the VM. Or do more through knowing the year like partitioning the list into a list of lists, each one being for a specific year. Which would again would be much nicer to happen in the VM itself. Then the UI can just take in the state and render it, instead of having to care about such logic.